### DISCUSSION TITLE
Enabling Java VisualVM for simulation?

### DISCUSSION THREAD

**Discussion by chauser** [wpilibsuite/2025Beta#68] [Q&A] [answered]
The Gradle setup instructions for enabling VisualVM place the settings in the roborio target section. What would be the build.gradle changes to support VisualVM for simulation?

**Comment by chauser** [comment]
It seems that configuration in build.gradle is not needed. Simply download the VSCode VisualVM extension and follow the instructions.

**Reply by Gold856** [reply, marked as answer]
Even simpler, you can actually open up VisualVM and profile a local Java process (including the robot program). No external tools required.

**Reply by chauser** [reply]
Thanks.
